frontend/public/module/k8s/autocomplete.ts

```diff
@@ -9,6 +9,72 @@ import { K8sKind, K8sResourceKind } from '../../module/k8s';
 import { k8sListPartialMetadata } from '../../module/k8s/resource';
 import { getActiveNamespace } from '../../ui/ui-actions';
 
+export type AceSnippet = {
+  content: string;
+  name?: string;
+  guard?: string;
+  trigger?: string;
+  endTrigger?: string;
+  endGuard?: string;
+  tabTrigger?: string;
+};
+
+export const snippets = ImmutableMap<string, string>()
+  .set('metadata', `
+    metadata:
+      name: \${1}
+      namespace: \${2}
+      labels: 
+        \${3}
+      annotations:
+        \${4}
+  `)
+  .set('selector', `
+    selector:
+      matchLabels:
+        \${1}
+  `)
+  .set('containers', `
+    containers:
+      - name: \${1}
+        image: \${2}
+        ports:
+          - containerPort: \${3}
+        volumeMounts:
+          - name: \${4}
+            mountPath: \${5}
+        command:
+          - \${6}
+  `)
+  .set('volumeClaimTemplates', `
+    volumeClaimTemplates:
+      - metadata:
+          name: \${1}
+        spec:
+          accessModes:
+            - ReadWriteOnce
+          storageClassName: \${2}
+          resources:
+            requests:
+              storage: 1Gi
+  `)
+  .set('rules', `
+    rules:
+      - http:
+          paths:
+            - path: /\${1}
+              backend:
+                serviceName: \${2}
+                servicePort: 80
+  `)
+  .set('ports', `
+    ports:
+    - protocol: TCP
+      port: 80
+      targetPort: \${1}
+  `)
+  .map((content, name) => ({name, content: content.split('\n').slice(1).map(s => s.substring(4)).join('\n')} as AceSnippet));
```

Comment:
Really like this these snippets :+1: 
Should this PR just contain basic set of snippets, and we add additional snippets on demand, eg:
``` yaml
parameters:
- name: {
  value: 
  description: 
```

or 

```yaml
triggers:
- imageChange:
    from:
      kind: ImageStreamTag
      name: hello-openshift-binary:latest
  type: ImageChange
```

or those are all we want to have ?

Reply:
@jhadvig Added those snippets for you. We can make follow-up PRs to add additional snippets as they are needed.